packages/@webex/plugin-cc/test/unit/spec/cc.ts

```diff
@@ -1,348 +1,248 @@
+        agentId: 'agent123',
+        agentMailId: '',
+        agentName: 'John',
+        teams: [],
+        agentProfileId: '',
+        loginVoiceOptions: [],
+        idleCodes: [],
+        wrapUpCodes: [],
+      };
 
-  describe('#addEventHandler', () => {
-    it('should add an event handler with timeout', () => {
-      jest.useFakeTimers();
-    
-      const resolveMock = jest.fn();
-      const rejectMock = jest.fn();
-    
-      webex.cc.addEventHandler('register', resolveMock, rejectMock);
-    
-      const eventHandler = webex.cc.eventHandlers.get('register');
-      expect(eventHandler).toBeDefined();
-      expect(eventHandler.resolve).toBe(resolveMock);
-      expect(eventHandler.reject).toBe(rejectMock);
-      expect(eventHandler.timeoutId).toBeDefined();
-    
-      expect(resolveMock).not.toHaveBeenCalled();
-    
-      // Fast-forward until all timers have been executed
-      jest.runAllTimers();
-    
-      expect(rejectMock).toHaveBeenCalled();
-      clearTimeout(eventHandler.timeoutId);
-    
-      jest.useRealTimers();
-    });
-  
-    // it('should reject the promise if the event times out', (done) => {
-    //   const resolveMock = jest.fn();
-    //   const rejectMock = jest.fn().mockImplementation((error) => {
-    //     expect(error.message).toBe('Time out waiting for event: register');
-    //     done();
-    //   });
-  
-    //   webex.cc.addEventHandler('register', resolveMock, rejectMock);
-  
-    //   const eventHandler = webex.cc.eventHandlers.get('register');
-    //   expect(eventHandler).toBeDefined();
-  
-    //   // Simulate timeout
-    //   setTimeout(() => {
-    //     expect(rejectMock).toHaveBeenCalled();
-    //   }, WEBSOCKET_EVENT_TIMEOUT + 100);
-    // }, WEBSOCKET_EVENT_TIMEOUT + 200);
-  });
+      mockAgentConfig.getAgentProfile.mockResolvedValue(mockAgentProfile);
 
-  describe('#establishConnection', () => {
-    it('should establish WebSocket connection with correct parameters', async () => {
-      webSocketMock.subscribeAndConnect.mockResolvedValue();
+      mockHttpRequest.subscribeNotifications.mockResolvedValue({
+        agentId: 'agent123',
+      });
 
-      webex.cc.register();
+      const result = await webex.cc.register();
 
-      expect(webSocketMock.subscribeAndConnect).toHaveBeenCalled();
-      expect(webSocketMock.subscribeAndConnect).toHaveBeenCalledWith({
-        datachannelUrl: 'https://api.example.com/v1/notification/subscribe',
+      expect(mockHttpRequest.subscribeNotifications).toHaveBeenCalledWith({
         body: {
           force: true,
           isKeepAliveEnabled: false,
           clientType: 'WebexCCSDK',
           allowMultiLogin: true,
         },
       });
-    }, 1000);
+      expect(mockAgentConfig.getAgentProfile).toHaveBeenCalled();
+      expect(webex.logger.log).toHaveBeenCalledWith(
+        'file: cc: agent config is fetched successfully'
+      );
+      expect(result).toEqual(mockAgentProfile);
```

Comment:
We should have a check(expect) for connectWebsocket function being invoked when register is invoked

Reply:
addressed